src/zarr/core/array.py

```diff
@@ -4006,31 +3998,34 @@ def _get_default_chunk_encoding_v3(
     else:
         dtype_key = "numeric"
 
-    codec_dicts = default_codecs[dtype_key]
-    codecs = tuple(_resolve_codec(c) for c in codec_dicts)
-    array_bytes_maybe = None
-    array_array: list[ArrayArrayCodec] = []
-    bytes_bytes: list[BytesBytesCodec] = []
+    default_filters = zarr_config.get("array.v3_default_filters").get(dtype_key)
+    default_serializer = zarr_config.get("array.v3_default_serializer").get(dtype_key)
+    default_compressors = zarr_config.get("array.v3_default_compressors").get(dtype_key)
 
-    for codec in codecs:
-        if isinstance(codec, ArrayBytesCodec):
-            if array_bytes_maybe is not None:
-                raise ValueError(
-                    f"Got two instances of ArrayBytesCodec: {array_bytes_maybe} and {codec}. "
-                    "Only one array-to-bytes codec is allowed."
-                )
-            array_bytes_maybe = codec
-        elif isinstance(codec, ArrayArrayCodec):
-            array_array.append(codec)
-        elif isinstance(codec, BytesBytesCodec):
-            bytes_bytes.append(codec)
-        else:
-            raise TypeError(f"Unexpected codec type: {type(codec)}")
+    filters_list: list[ArrayArrayCodec] = []
+    compressors_list: list[BytesBytesCodec] = []
 
-    if array_bytes_maybe is None:
+    serializer = _resolve_codec(default_serializer)
```

Comment:
can we use [`_parse_bytes_bytes_codec`](https://github.com/zarr-developers/zarr-python/blob/617e2cd84b28c4ca28b8b221abd77718561556bb/src/zarr/registry.py#L169-L186) here? (and similarly for the other codec varieties)

Reply:
Good idea. That removes some lines of code.